flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -254,11 +230,24 @@
 	public static final ConfigOption<String> TOTAL_PROCESS_MEMORY =
 		key("taskmanager.memory.total-process.size")
 			.noDefaultValue()
-			.withDeprecatedKeys(TASK_MANAGER_HEAP_MEMORY.key())
+			.withDeprecatedKeys("taskmanager.heap.size")
 			.withDescription("Total Process Memory size for the TaskExecutors. This includes all the memory that a"
 				+ " TaskExecutor consumes, consisting of Total Flink Memory, JVM Metaspace, and JVM Overhead. On"
 				+ " containerized setups, this should be set to the container memory.");
 
+	/**
+	 * JVM heap size (in megabytes) for the TaskManagers.
+	 *
+	 * @deprecated use {@link #TOTAL_PROCESS_MEMORY}
+	 */
+	@Deprecated
+	public static final ConfigOption<Integer> TASK_MANAGER_HEAP_MEMORY_MB =
```

Comment:
could we avoid reordering of options to keep git history?

Reply:
I think this way it looks more clear to put deprecated option close to the new alternative.